internal/v2/cert_verifier/cert_verifier.go

```diff
@@ -5,20 +5,25 @@ import (
 	"fmt"
 	"crypto/x509"
 	"google.golang.org/grpc/codes"
+	"github.com/google/s2a-go/internal/tokenmanager"
 
 	s2av2pb "github.com/google/s2a-go/internal/proto/v2/s2a_go_proto"
 )
 
 // VerifyClientCertificateChain builds a SessionReq, sends it to S2Av2 and
 // receives a SessionResp.
-func VerifyClientCertificateChain(cstream s2av2pb.S2AService_SetUpSessionClient) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+func VerifyClientCertificateChain(cstream s2av2pb.S2AService_SetUpSessionClient, tokenManager tokenmanager.AccessTokenManager) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
 	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		// Get Access Token
+		token, err := tokenManager.DefaultToken()
```

Comment:
Super sorry I missed this on the first pass, I forgot that the plan had changed here. We don't need to attach an auth mechanism in the cert verifier or the remote signer, only in the TLS config store.

Reply:
No worries, Thanks for letting me know! This is because S2A only looks at the token attached to the first SessionReq sent on the stream right, and TLS config store is always going to make the first SessionReq?